File: app/src/main/java/com/absinthe/anywhere_/utils/timber/ReleaseTree.java
package com.absinthe.anywhere_.utils.timber;

import android.util.Log;

import com.absinthe.anywhere_.model.GlobalValues;

import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

import timber.log.Timber;

public class ReleaseTree extends Timber.DebugTree {

    @Override
    protected boolean isLoggable(@Nullable String tag, int priority) {
        return !(priority == Log.VERBOSE || priority == Log.DEBUG || priority == Log.INFO)
                || GlobalValues.sIsDebugMode;
    }

    @Override
    protected void log(int priority, @Nullable String tag, @NotNull String message, @Nullable Throwable t) {
        if (!isLoggable(tag, priority)) {
            return;
        }
        super.log(priority, tag, message, t);
    }

}

File: app/src/main/java/com/absinthe/anywhere_/utils/timber/ReleaseTree.kt
package com.absinthe.anywhere_.utils.timber

import android.util.Log
import com.absinthe.anywhere_.constants.GlobalValues
import timber.log.Timber.DebugTree

class ReleaseTree : DebugTree() {

  override fun isLoggable(tag: String?, priority: Int): Boolean {
    return (!(priority == Log.VERBOSE || priority == Log.DEBUG || priority == Log.INFO)
      || GlobalValues.sIsDebugMode)
  }

  override fun log(priority: Int, tag: String?, message: String, t: Throwable?) {
    if (!isLoggable(tag, priority)) {
      return
    }
    super.log(priority, tag, message, t)
  }
}
